src/logic/safe/transactions/gas.ts

```diff
@@ -217,6 +159,97 @@ export const estimateExecTransactionGas = async (
     )
   } catch (error) {
     console.info('Error calculating tx gas estimation', error.message)
+    throw error
+  }
+}
+
+type TransactionExecutionEstimationProps = {
+  txData: string
+  safeAddress: string
+  txRecipient: string
+  txConfirmations?: List<Confirmation>
+  txAmount: string
+  operation: number
+  gasPrice: string
+  gasToken: string
+  refundReceiver: string // Address of receiver of gas payment (or 0 if tx.origin).
+  safeTxGas: number
+  from: string
+  approvalAndExecution?: boolean
+}
+
+export const estimateGasForTransactionExecution = async ({
+  safeAddress,
+  txRecipient,
+  txConfirmations,
+  txAmount,
+  txData,
+  operation,
+  gasPrice,
+  gasToken,
+  refundReceiver,
+  safeTxGas,
+  approvalAndExecution,
+}: TransactionExecutionEstimationProps): Promise<number> => {
+  const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+  try {
+    if (approvalAndExecution) {
```

Comment:
If we try to use `execTransaction .call()` to estimate the gas without having all the "confirmed" signatures (no prevalidated with the sentinel) we got an error: `Hash has not been approved`, so in case that the user wants to do an "approve and execute" we will calculate the gas using the same method as for creation `safe.requireTransactionGas`.